src/lib/operators/index_scan.cpp

```diff
@@ -34,175 +34,111 @@ const std::string& IndexScan::name() const {
 std::shared_ptr<const Table> IndexScan::_on_execute() {
   _in_table = left_input_table();
 
-  _validate_input();
+  auto chunk_id_mapping = std::unordered_map<ChunkID, ChunkID>{};
+  auto pruned_chunk_ids_set = std::unordered_set<ChunkID>{};
 
-  _out_table = std::make_shared<Table>(_in_table->column_definitions(), TableType::References);
+  // If the input operator is of the type GetTable, pruning must be considered.
+  const auto& input_get_table = dynamic_pointer_cast<const GetTable>(left_input());
+  Assert(input_get_table, "IndexScan needs a GetTable operator as input.");
 
-  std::mutex output_mutex;
+  // If columns have been pruned, calculate the ColumnID that was originally indexed.
+  const auto& pruned_column_ids = input_get_table->pruned_column_ids();
+  _indexed_column_id = column_id_before_pruning(_indexed_column_id, pruned_column_ids);
 
-  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
-  if (included_chunk_ids.empty()) {
-    const auto chunk_count = _in_table->chunk_count();
-    jobs.reserve(chunk_count);
-    for (auto chunk_id = ChunkID{0u}; chunk_id < chunk_count; ++chunk_id) {
-      const auto chunk = _in_table->get_chunk(chunk_id);
-      Assert(chunk, "Physically deleted chunk should not reach this point, see get_chunk / #1686.");
+  // If chunks have been pruned, calculate a mapping that maps the pruned ChunkIDs to the original ones.
+  const auto& pruned_chunk_ids = input_get_table->pruned_chunk_ids();
+  chunk_id_mapping = chunk_ids_after_pruning(_in_table->chunk_count() + pruned_chunk_ids.size(), pruned_chunk_ids);
+  pruned_chunk_ids_set = std::unordered_set<ChunkID>{pruned_chunk_ids.begin(), pruned_chunk_ids.end()};
 
-      jobs.push_back(_create_job(chunk_id, output_mutex));
-    }
-  } else {
-    jobs.reserve(included_chunk_ids.size());
-    for (const auto chunk_id : included_chunk_ids) {
-      if (_in_table->get_chunk(chunk_id)) {
-        jobs.push_back(_create_job(chunk_id, output_mutex));
-      }
+  // Remove all values from the mapping not present in the included ChunkIDs.
+  for (auto iterator = chunk_id_mapping.begin(); iterator != chunk_id_mapping.end();) {
```

Comment:
I also don't really like it; if there are more elegant solutions, please tell me :)

Reply:
I don't think removing the not included ChunkIDs from the mapping is necessary since only pruned and not indexed ChunkIDs won't be included by the LQPTranslator.
Not-indexed ChunkIDs won't make it into the mapping, and not-indexed ChunkIDs are irrelevant here.